frontend/lib/src/components/shared/InputInstructions/InputInstructions.tsx

```diff
@@ -51,7 +53,10 @@ const InputInstructions = ({
     )
   }
 
-  if (dirty) {
+  // Show enter instruction if not a form or form allows submit on Enter
+  const showEnterInstruction = allowSubmitOnEnter || !inForm
```

Comment:
nit: If `allowSubmitOnEnter` is False and not in a form, it would still show the enter instruction -> this is a bit confusing based on the parameter name. Can't we make `allowSubmitOnEnter` the ultimate decider to show or not to show the enter instruction and not involve `inForm` here? It would probably need to move this check into the components like:

```typescript
const allowSubmitOnEnter = !isInForm(formId) || widgetMgr.allowFormSubmitOnEnter(formId)
```

I think this makes the behavior more explicit. An alternative would be to use a more specific name like `allowFormSubmitOnEnter`.